### ISSUE TITLE
Hive Connection Error with ZooKeeper

### ISSUE THREAD

**Issue by mwaqashraf** [dbeaver/dbeaver#22777]
### Description

Hi, 

I created new connection to hive and connecting with zookeeper quorum, it is throwing below error. When I create direct connection to hive server, it is working fine. Can you please help fixing this error?

Connection URL (Working Fine):

jdbc:hive2://host:10000/db_name

Connection URL (Throwing Error):

jdbc:hive2://hostname:2181/;serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=hiveserver2

Unexpected driver error occurred while connecting to the database
'com.google.common.util.concurrent.ListeningExecutorService com.google.common.util.concurrent.MoreExecutors.sameThreadExecutor()'
'com.google.common.util.concurrent.ListeningExecutorService com.google.common.util.concurrent.MoreExecutors.sameThreadExecutor()'

### DBeaver Version

Enterprise Version 23.3.0.202312121701

### Operating System

Windows 11

### Database and driver

Database Name: Apache Hive
Driver: hive-jdbc-uber-2.6.5.0-292.jar

### Steps to reproduce

1. Create new connection with Apache Hive
2. give below URL (Connect through ZooKeeper)
jdbc:hive2://hostname:2181/;serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=hiveserver2
3. Test connection, it will throw the error

### Additional context

When connection created direct to Hive, it would work fine.
jdbc:hive2://hostname:10000/db_name

**Comment by E1izabeth** [comment]
Please, provide the log files. You can find instructions here https://dbeaver.com/docs/wiki/Log-files/.

Also how are you configuring connection: by host or by url?
![image](https://github.com/dbeaver/dbeaver/assets/28875055/70bb6c6e-a2e1-47e3-98e5-ffa0126de64c)

**Comment by mwaqashraf** [comment]
[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/14232028/dbeaver-debug.log)

Attached debug log, I am configuring connection by URL.

**Comment by mwaqashraf** [comment]
Please note that this error is not coming in older DBeaver EE versions tested on 23.2.* an d23.0.*. Latest version 23.3.* is impacted

**Comment by E1izabeth** [comment]
Thank you.
Error from log:
```
ava.lang.NoSuchMethodError: 'com.google.common.util.concurrent.ListeningExecutorService com.google.common.util.concurrent.MoreExecutors.sameThreadExecutor()'
	at org.apache.curator.framework.listen.ListenerContainer.addListener(ListenerContainer.java:40)
	at org.apache.curator.framework.imps.CuratorFrameworkImpl.start(CuratorFrameworkImpl.java:254)
	at org.apache.hive.jdbc.ZooKeeperHiveClientHelper.getZkClient(ZooKeeperHiveClientHelper.java:61)
	at org.apache.hive.jdbc.ZooKeeperHiveClientHelper.configureConnParams(ZooKeeperHiveClientHelper.java:107)
	at org.apache.hive.jdbc.Utils.configureConnParams(Utils.java:526)
	at org.apache.hive.jdbc.Utils.parseURL(Utils.java:446)
	at org.apache.hive.jdbc.HiveConnection.<init>(HiveConnection.java:185)
	at org.apache.hive.jdbc.HiveDriver.connect(HiveDriver.java:105)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCConnectionOpener.run(JDBCConnectionOpener.java:109)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCConnectionOpener.run(JDBCConnectionOpener.java:83)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.openConnection(JDBCDataSource.java:221)
	at org.jkiss.dbeaver.ext.generic.model.GenericDataSource.openConnection(GenericDataSource.java:158)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCExecutionContext.connect(JDBCExecutionContext.java:103)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.initializeMainContext(JDBCRemoteInstance.java:102)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.<init>(JDBCRemoteInstance.java:61)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.initializeRemoteInstance(JDBCDataSource.java:119)
	at org.jkiss.dbeaver.ext.generic.model.GenericDataSource.<init>(GenericDataSource.java:123)
	at org.jkiss.dbeaver.ext.hive.model.HiveDataSource.<init>(HiveDataSource.java:45)
	at org.jkiss.dbeaver.ext.hive.model.HiveMetaModel.createDataSourceImpl(HiveMetaModel.java:51)
	at org.jkiss.dbeaver.ext.generic.GenericDataSourceProvider.openDataSource(GenericDataSourceProvider.java:106)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.openDataSource(DataSourceDescriptor.java:1288)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect0(DataSourceDescriptor.java:1125)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect(DataSourceDescriptor.java:950)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.run(ConnectJob.java:77)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)

```

**Comment by mwaqashraf** [comment]
Hi,

wanted to check if there is any udpate.

**Comment by E1izabeth** [comment]
No update yet

**Comment by E1izabeth** [comment]
@mwaqashraf Could you try changing the driver for Hive to this https://mvnrepository.com/artifact/org.apache.hive/hive-jdbc/4.0.0-beta-1?

**Comment by mwaqashraf** [comment]
Hi @E1izabeth,

Getting below error, attached debug logs.

org.apache.thrift.protocol.TProtocol
[Uploading dbeaver-debug.log…]()

**Comment by mwaqashraf** [comment]
[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/14511474/dbeaver-debug.log)

**Comment by E1izabeth** [comment]
Thank you

**Comment by E1izabeth** [comment]
```
2024-03-06 17:51:37.348 - Execution context opened
2024-03-06 17:51:37.415 - org.apache.thrift.protocol.TProtocol
java.lang.ClassNotFoundException: org.apache.thrift.protocol.TProtocol
	at java.base/java.net.URLClassLoader.findClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at org.apache.hive.jdbc.HiveDriver.connect(HiveDriver.java:94)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCConnectionOpener.run(JDBCConnectionOpener.java:109)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCConnectionOpener.run(JDBCConnectionOpener.java:83)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.openConnection(JDBCDataSource.java:221)
	at org.jkiss.dbeaver.ext.generic.model.GenericDataSource.openConnection(GenericDataSource.java:158)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCExecutionContext.connect(JDBCExecutionContext.java:103)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.initializeMainContext(JDBCRemoteInstance.java:102)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCRemoteInstance.<init>(JDBCRemoteInstance.java:61)
	at org.jkiss.dbeaver.model.impl.jdbc.JDBCDataSource.initializeRemoteInstance(JDBCDataSource.java:119)
	at org.jkiss.dbeaver.ext.generic.model.GenericDataSource.<init>(GenericDataSource.java:123)
	at org.jkiss.dbeaver.ext.hive.model.HiveDataSource.<init>(HiveDataSource.java:45)
	at org.jkiss.dbeaver.ext.hive.model.HiveMetaModel.createDataSourceImpl(HiveMetaModel.java:51)
	at org.jkiss.dbeaver.ext.generic.GenericDataSourceProvider.openDataSource(GenericDataSourceProvider.java:106)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.openDataSource(DataSourceDescriptor.java:1288)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect0(DataSourceDescriptor.java:1125)
	at org.jkiss.dbeaver.registry.DataSourceDescriptor.connect(DataSourceDescriptor.java:950)
	at org.jkiss.dbeaver.runtime.jobs.ConnectJob.run(ConnectJob.java:77)
	at org.jkiss.dbeaver.runtime.jobs.ConnectionTestJob.run(ConnectionTestJob.java:103)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

**Comment by ShadelessFox** [comment]
Hello @mwaqashraf,

Since we don't have a suitable environment for reproducing the issue, can you try adding the missing dependency yourself to see if it works?

Open **Database** -> **Driver Manager** and follow steps shown on the screenshot:
![image](https://github.com/dbeaver/dbeaver/assets/35821147/572a863b-8724-4d0d-92a1-a529ffc544e7)

Dependency:
```xml
<dependency>
    <groupId>org.apache.thrift</groupId>
    <artifactId>libthrift</artifactId>
    <version>0.19.0</version>
</dependency>
```

If it doesn't work, try this one instead:
```
<dependency>
    <groupId>org.apache.thrift</groupId>
    <artifactId>libthrift</artifactId>
    <version>0.11.0</version>
</dependency>
```

**Comment by mwaqashraf** [comment]
Now getting this error with both dependancies.

org.apache.hive.service.rpc.thrift.TCLIService$Iface

**Comment by mwaqashraf** [comment]
[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/14548924/dbeaver-debug.log)

**Comment by ShadelessFox** [comment]
Try adding this dependency instead of the previous one (delete the `org.apache.thrift:libthrift` first):

```xml
<dependency>
    <groupId>org.apache.hive</groupId>
    <artifactId>hive-service-rpc</artifactId>
    <version>3.1.3</version>
</dependency>
```

And make sure the "Load optional dependencies" checkbox is set.

**Comment by mwaqashraf** [comment]
this one coming now.
org.apache.hive.service.cli.HiveSQLException

[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/14584322/dbeaver-debug.log)

**Comment by ShadelessFox** [comment]
We're going somewhere. How deep is the rabbit hole?

Try removing **all** dependencies and add this one instead:
```xml
<dependency>
    <groupId>org.apache.hive</groupId>
    <artifactId>hive-jdbc</artifactId>
    <version>3.1.3</version>
</dependency>
```

The driver we ship is quite outdated and missing a set of dependencies. Apache manages this one, and it should be up to date.

**Comment by mwaqashraf** [comment]
no good news.

Error: HIVE_DEFAULT_NULLS_LAST
Moreover, one depency can't be downloaded "Can't open 'https://repo1.maven.org/maven2/org/apache/curator/apache-curator/2.12.0/apache-curator-2.12.0.jar': Not Found"

[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/14597224/dbeaver-debug.log)

**Comment by linghengqian** [comment]
- While investigating https://github.com/apache/shardingsphere/pull/31526 , I noticed that DBeaver was unable to connect to HiveServer2 due to missing dependencies. This appears to involve an undocumented breaking change in apache/hive 4.0.0, hence I opened https://issues.apache.org/jira/browse/HIVE-28308 to address the issue.

**Comment by E1izabeth** [comment]
@linghengqian thank you

**Comment by linghengqian** [comment]
# TL; DR
- @E1izabeth  I found this a bit odd, Dbeaver only seems to be able to configure one Maven dependency for one JDBC driver? I'm not sure if I'm missing something in the Dbeaver documentation.
- First, start a HiveServer2 Docker Container instance through the bash command `docker run -d -p 10000:10000 -p 10002:10002 --env SERVICE_NAME=hiveserver2 apache/hive:4.0.0`. For Remote HiveServer2 connected to apache/hive:4.0.0, this requires two Maven dependencies. I simply exclude multiple slf4j-implemented log libraries in `org.apache.hadoop:hadoop-common:3.3.6` here. Apparently you should not use `org.apache.hadoop:hadoop-common:3.4.0`, apache/hadoop:3.4.0 has never been compatible with apache/hive, see https://github.com/apache/hive/pull/5187 .
```xml
<dependencies>
        <dependency>
            <groupId>org.apache.hive</groupId>
            <artifactId>hive-jdbc</artifactId>
            <version>4.0.0</version>
            <classifier>standalone</classifier>
        </dependency>
        <dependency>
            <groupId>org.apache.hadoop</groupId>
            <artifactId>hadoop-common</artifactId>
            <version>3.3.6</version>
            <exclusions>
                <exclusion>
                    <groupId>org.slf4j</groupId>
                    <artifactId>slf4j-reload4j</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
</dependencies>
```
- According to the clarification of https://issues.apache.org/jira/browse/HIVE-28418 , `org.apache.hive:hive-jdbc` with `classifier` as `standalone` can only connect to Remote HiveServer2, not Embedded HiveServer2.
- Connect to the Remote HiveServer2 just created via the `jdbcUrl` of `jdbc:hive2://localhost:10000/` and the `DriverClassName` of `org.apache.hive.jdbc.HiveDriver`. Execute some random SQL.
```sql
CREATE DATABASE demo_ds_0;
use demo_ds_0;
drop table if exists testHiveDriverTable;
create table testHiveDriverTable (key int, value string);
show tables 'testHiveDriverTable';
describe testHiveDriverTable;
```
- I used DataGrip under WSL for verification. Because I really can't seem to configure Dbeaver to use multiple Maven dependencies for HiveServer2 JDBC Driver.
- ![image](https://github.com/user-attachments/assets/38e5ff47-d4eb-4a36-bb97-19f53ec3c418)
- ![image](https://github.com/user-attachments/assets/8d73cd4c-2b2b-4ac7-8f1b-29922db5a358)
# What happened in the master branch of apache/hive?
- The need to introduce `org.apache.hadoop:hadoop-common:3.3.6` is indeed a bug in apache/hive, which has been fixed in https://github.com/apache/hive/pull/5313 and https://issues.apache.org/jira/browse/HIVE-28315 . You can compile apache/hive manually first. Compilation was done with SDKMAN! under Ubuntu 22.04.4. I would say it's a huge shame that apache/hive can only be compiled with OpenJDK8 so far. Feel free to add some PRs to https://issues.apache.org/jira/browse/HIVE-26473 .
```shell
sdk install java 8.0.422-tem
sdk use java 8.0.422-tem
sdk install maven
git clone git@github.com:apache/hive.git
cd ./hive/
git reset --hard 929305f077f5e40311939ce7256adb3b35d985b0
mvn clean install -DskipTests -T1C 
mvn clean package -pl packaging -DskipTests -Pdocker
cd ../
```
- After `docker run -d -p 10000:10000 -p 10002:10002 --env SERVICE_NAME=hiveserver2 apache/hive:4.1.0-SNAPSHOT`, only the following dependencies are required.
```xml
<dependency>
            <groupId>org.apache.hive</groupId>
            <artifactId>hive-jdbc</artifactId>
            <version>4.1.0-SNAPSHOT</version>
            <classifier>standalone</classifier>
</dependency>
```

**Comment by linghengqian** [comment]
- What happened with dbeaver is a bit of a blind spot for me. I don't know what fonts dbeaver's i18n relies on.
- ![image](https://github.com/user-attachments/assets/656c3824-a599-498f-bec3-c30351edadde)
```shell
sudo apt update && sudo apt upgrade -y
sudo apt-get install fonts-noto-cjk fonts-noto-color-emoji -y
sudo snap install dbeaver-ce
snap run dbeaver-ce
```
- Overall, it's pretty normal from my perspective. Although I don't know why dbeaver recognizes HiveServer2 of apache/hive:4.0.0 as 3.3.6.
- ![image](https://github.com/user-attachments/assets/84190396-9899-45ba-9064-248c678bc249)
- ![image](https://github.com/user-attachments/assets/da1fd403-05bd-4eff-b660-73a28d6b1b85)
- ![image](https://github.com/user-attachments/assets/6eda0371-d324-4a7b-82d1-1911c158958d)

**Comment by E1izabeth** [comment]
Thank you for investigation.
Regarding the fonts, you can check, which font is used and change it in preferences. As far as I see, it's a Banner font and/or Header font.
![image](https://github.com/user-attachments/assets/8d928a03-3b96-45ac-9a1b-2c0a69dbaa86)

**Comment by ShadelessFox** [comment]
@E1izabeth window titles have nothing to do with the chosen font within the app. More likely an encoding issue within Eclipse.

**Comment by linghengqian** [comment]
> Thank you for investigation. Regarding the fonts, you can check, which font is used and change it in preferences. As far as I see, it's a Banner font and/or Header font. 

- @E1izabeth I made an update at https://github.com/dbeaver/dbeaver/issues/35028#issuecomment-2260835536 to stop the current issue from going off topic.
- Starting a HiveServer2 that uses a zookeeper server as the service discovery of the Hive Metastore usually requires the use of Docker Compose, as this requires multiple containers. Apparently there is no embedded zookeeper server in the Docker Image of HiveServer2, and a separate docker image is required for the zookeeper server. 
- The current time is a bit late for people in the East 8th zone, I will observe tomorrow whether there is any problem using dbeaver to connect to a HiveServer2 that uses a zookeeper server as the service discovery. This is documented at https://cwiki.apache.org/confluence/display/Hive/HiveServer2+Clients#HiveServer2Clients-ConnectionURLWhenZooKeeperServiceDiscoveryIsEnabled .

**Comment by E1izabeth** [comment]
@linghengqian thank you

**Comment by linghengqian** [comment]
## TL:DR
- Just set HiveServer2 JDBC Driver in dbeaver to use the following two dependencies. Although `org.apache.hadoop:hadoop-common:3.3.6` has some internal dependency conflict.
```xml
<dependency>
            <groupId>org.apache.hive</groupId>
            <artifactId>hive-jdbc</artifactId>
            <version>4.0.0</version>
            <classifier>standalone</classifier>
</dependency>
<dependency>
            <groupId>org.apache.hadoop</groupId>
            <artifactId>hadoop-common</artifactId>
            <version>3.3.6</version>
</dependency>
```
## The simplest verification
- Make sure the host ports 2181, 10000, and 10002 are available. 
- Define the Docker Compose unit file. Set HiveServer2's container `hostname` to `127.0.0.1` to facilitate calling from the local environment outside the docker network of this docker compose unit. Apparently this is an interesting limitation of docker.
```yaml
services:
  zookeeper-server:
    image: zookeeper:3.9.2-jre-17
    restart: always
    ports:
      - "2181:2181"
  hive-server2:
    image: apache/hive:4.0.0
    restart: always
    hostname: '127.0.0.1'
    depends_on:
      zookeeper-server:
        condition: service_started
    environment:
      SERVICE_NAME: hiveserver2
      HIVE_CUSTOM_CONF_DIR: /hive_custom_conf
    ports:
      - "10000:10000"
      - "10002:10002"
    volumes:
      - ./hive-custom-conf:/hive_custom_conf
```
- Define a `hive-site.xml` in the `hive-custom-conf` directory at the same level so that it can be mounted into the Docker Container of HiveServer2. Only `hive.server2.support.dynamic.service.discovery` and `hive.zookeeper.quorum` need to be set manually. Others are the original Docker Container configurations.
```xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <property>
        <name>hive.server2.enable.doAs</name>
        <value>false</value>
    </property>
    <property>
        <name>hive.tez.exec.inplace.progress</name>
        <value>false</value>
    </property>
    <property>
        <name>hive.tez.exec.print.summary</name>
        <value>true</value>
    </property>
    <property>
        <name>hive.exec.scratchdir</name>
        <value>/opt/hive/scratch_dir</value>
    </property>
    <property>
        <name>hive.user.install.directory</name>
        <value>/opt/hive/install_dir</value>
    </property>
    <property>
        <name>tez.runtime.optimize.local.fetch</name>
        <value>true</value>
    </property>
    <property>
        <name>hive.exec.submit.local.task.via.child</name>
        <value>false</value>
    </property>
    <property>
        <name>mapreduce.framework.name</name>
        <value>local</value>
    </property>
    <property>
        <name>tez.local.mode</name>
        <value>true</value>
    </property>
    <property>
        <name>hive.metastore.warehouse.dir</name>
        <value>/opt/hive/data/warehouse</value>
    </property>
    <property>
        <name>metastore.metastore.event.db.notification.api.auth</name>
        <value>false</value>
    </property>

    <property>
        <name>hive.server2.support.dynamic.service.discovery</name>
        <value>true</value>
    </property>
    <property>
        <name>hive.zookeeper.quorum</name>
        <value>zookeeper-server:2181</value>
    </property>
</configuration>
```
- Start the Docker Compose unit. We can now see in Portainer that the Compose unit has started.
- ![image](https://github.com/user-attachments/assets/22d1e627-4c89-42ed-8492-ccb77b2b6f6f)
- Use ZooNavigator to connect to the zookeeper server at localhost:2181 . This is mainly for detection. At this time, the content `hive.server2.instance.uri=localhost:10000;hive.server2.authentication=NONE;hive.server2.transport.mode=binary;hive.server2.thrift.sasl.qop=auth;hive.server2.thrift.bind.host=localhost;hive.server2.thrift.port=10000;hive.server2.use.SSL=false` exists under the zookeeper node of `/hiveserver2/serverUri=localhost:10000;version=4.0.0;sequence=0000000000`. This indicates that HiveServer2 has successfully enabled Zookeeper Service Discovery.
- ![image](https://github.com/user-attachments/assets/715dbd75-2f00-4cec-92e6-5d4d4e516b00)
- Open dbeaver and set the HiveServer2 JDBC Driver.
- ![image](https://github.com/user-attachments/assets/867c4f9c-c7cd-495a-af96-7d21dd4b6e97)

- Now in dbeaver, I can connect to HiveServer2 normally through the jdbcUrl of `jdbc:hive2://localhost:2181/;serviceDiscoveryMode=zooKeeper;zooKeeperNamespace=hiveserver2;`. The default value of the `zooKeeperNamespace` parameter is `hiveserver2` and can be omitted. The unfortunate point is that dbeaver identifies the driver as Hive JDBC 3.3.6. It is actually Hive JDBC 4.0.0.
- ![image](https://github.com/user-attachments/assets/d8786117-4cd1-48c2-a526-1a065d52fdb2)
- Feel free to test some SQL. No problem.
```sql
CREATE DATABASE demo_ds_0;
use demo_ds_0;
drop table if exists testHiveDriverTable;
create table testHiveDriverTable (key int, value string);
show tables 'testHiveDriverTable';
describe testHiveDriverTable;
```
- ![image](https://github.com/user-attachments/assets/8bcf320e-8173-4653-953a-ca28403de42f)

- You can also verify this using the unit test I created at https://github.com/linghengqian/hivesever2-v400-sd-test .

```shell
sdk install java 22.0.2-graalce
sdk use java 22.0.2-graalce
git clone git@github.com:linghengqian/hivesever2-v400-sd-test.git
cd ./hivesever2-v400-sd-test/
docker compose -f ./docker-compose-lingh.yml pull
docker compose -f ./docker-compose-lingh.yml up -d

# ... Wait five seconds for HiveServer2 to finish initializing.

./mvnw clean test
docker compose -f ./docker-compose-lingh.yml down
```

# External discussion around the current issue

- There is indeed an external discussion about the current issue. Feel free to check it out if you are interested.
    - https://issues.apache.org/jira/browse/HIVE-28295 , the embedded HiveServer2 is no longer encouraged to use. Users should start a Remote HiveServer2 or Remote Hive MetaStore through Docker Image.
    - https://issues.apache.org/jira/browse/HIVE-28308 , the old version of HiveServer2 JDBC Driver displays inaccurate dependencies on the Maven Central search site.
    - https://issues.apache.org/jira/browse/HIVE-28418 and https://github.com/apache/hive/pull/5313 , the classes of `org.apache.hadoop:hadoop-common:3.3.6` are not included in `org.apache.hive:hive-jdbc:4.0.0-standalone`.
    - https://issues.apache.org/jira/browse/HIVE-28423 , Hive documentation is missing description for `hive.server2.support.dynamic.service.discovery`.
    - https://issues.apache.org/jira/browse/HIVE-28424 , the Hive documentation lacks instructions on how to enable Zookeeper Service Discovery for HiveServer2 in the Docker Image.
    - https://issues.apache.org/jira/browse/HIVE-28429, using HiveServer2 JDBC Driver in OpenJDK 22 requires `org.apache.hadoop:hadoop-client-api:3.3.6`.
    - https://issues.apache.org/jira/browse/HIVE-28430 , if we need to modify the system properties of the Docker Image of HiveServer2 and Hive Metastore, we always need to repeat a part of the property configuration in `hive-site.xml`.
    - https://github.com/apache/hive/pull/5187, apache/hive has never been compatible with apache/hadoop:3.4.0 .

# What is the way forward for the current issue?
- I don't know who is preparing PR for the current issue. But to be honest, I am really a dbeaver newbie.

**Comment by kseniaguzeeva** [comment]
We are going to add a new Hive driver. The issue is going to be solved there.

**Comment by mwaqashraf** [comment]
I started using latest Cloudera JDBC driver for Hive which connects with zookeeper successfully.

**Comment by kseniaguzeeva** [comment]
The issue is fixed [here](https://github.com/dbeaver/pro/issues/3513) and it will be available in the next release.
